packages/rum/test/viewCollection.spec.ts

```diff
@@ -81,47 +81,62 @@ function spyOnViews() {
 
 describe('rum track url change', () => {
   let setupBuilder: TestSetupBuilder
-  let initialView: string
+  let initialViewId: string
   let initialLocation: Location
 
   beforeEach(() => {
     const fakeLocation: Partial<Location> = { pathname: '/foo' }
     mockHistory(fakeLocation)
-    setupBuilder = setup().withViewCollection(fakeLocation)
-    setupBuilder.build()
-    initialView = viewContext.id
-    initialLocation = viewContext.location
+    setupBuilder = setup()
+      .withViewCollection(fakeLocation)
+      .beforeBuild((lifeCycle) => {
+        const subscription = lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, (viewContext) => {
+          initialViewId = viewContext.id
+          initialLocation = viewContext.location
+          subscription.unsubscribe()
+        })
+      })
   })
 
   afterEach(() => {
     setupBuilder.cleanup()
   })
 
-  it('should update view id on path change', () => {
-    history.pushState({}, '', '/bar')
+  it('should create new view on path change', () => {
+    const { lifeCycle } = setupBuilder.build()
 
-    expect(viewContext.id).not.toEqual(initialView)
-    expect(viewContext.location).not.toEqual(initialLocation)
+    lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, (viewContext) => {
+      expect(viewContext.id).not.toEqual(initialViewId)
+      expect(viewContext.location).not.toEqual(initialLocation)
+    })
```

Comment:
This is a bit risky because if the event isn't triggered synchronously during the test, no assertion will be done. Maybe you can use a `done` callback to the test even if itsn't async?
```suggestion
    lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, (viewContext) => {
      expect(viewContext.id).not.toEqual(initialViewId)
      expect(viewContext.location).not.toEqual(initialLocation)
      done()
    })
```

Reply:
> Maybe you can use a done callback to the test even if itsn't async?

since the events are triggered synchronously, what is the reasoning for doing that?

Reply:
Because you are the author, you know this event will be triggered synchronously when doing a `pushState`. But this is not obvious.
* there is no test to show this. We could write a new test case to ensure this (with a spy for example)
* even then, it would help the reader to understand when the test case is ending

Reply:
From my understanding, your issue is that the test is written like an asynchronous test but it is not asynchronous.
IMO, this approach is similar to something like:
```js
it('...', () => {
  const callback = (a) => expect(a).toBe(true)
  testSubject(callback)
})
```
I don't think that we should write tests to ensure that this kind of code is synchronous.

I agree that using a spy would make the test easier to read and would prevent expectations not being run if there is any issue.